package/gluon-core/luasrc/lib/gluon/upgrade/820-dns-config

```diff
@@ -25,22 +25,33 @@ if dns.servers then
 	})
 end
 
+
+local function set_dns_record(name, host, sectionname)
+		uci:section('dhcp', 'domain', sectionname, {
+			name = name,
+			ip = host,
+		})
+end
+
+-- remove nextnode4_* and nextnode6_*
+uci:foreach("dhcp", "domain", function(s)
+	for key, value in pairs(s) do
+		if (key == ".name" and string.gmatch(value, "nextnode[46].*")) then
+			uci:delete("dhcp", value)
+		end
+	end
+end)
+
 if next_node.name and next_node.ip4 then
-	uci:section('dhcp', 'domain', 'nextnode4', {
-		name = next_node.name,
-		ip = next_node.ip4,
-	})
-else
-	uci:delete('dhcp', 'domain', 'nextnode4')
+	for i,n in ipairs(next_node.name) do
+		set_dns_record(n, next_node.ip4, 'nextnode4_' .. i)
+	end
 end
 
 if next_node.name and next_node.ip6 then
-	uci:section('dhcp', 'domain', 'nextnode6', {
-		name = next_node.name,
-		ip = next_node.ip6,
-	})
-else
-	uci:delete('dhcp', 'domain', 'nextnode6')
+	for i,n in ipairs(next_node.name) do
+		set_dns_record(n, next_node.ip6, 'nextnode6_' .. i)
+	end
```

Comment:
How about using a single loop for IPv4 and IPv6, just guarded by
```lua
if next_node.name then
```
? The ip4/ip6 checks could be moved into *set_dns_record* as `if not host`

Reply:
name was not checked at all before because it is not mandatory I guess. We could introduce a check that, if it is there checks if it is a table.

Reply:
host would be set in the ipv6 and in the ipv4 case, how would that help to distinguis? Also that would mean setting the section would have to move to yet another function or be duplicated due to the section name containing nextnoce4 or 6 respectively. Frankly I like this two-loop-approach more because I find it clearer. It may be a matter of taste though and if you really like the one-loop approach, I will adapt.

Reply:
Site checks are required even for optional fields, they are supposed to ensure that our update scripts never fail.

`host` is not set in both cases, the host field holds the IP address; this is confusing, lets rename it. I propose this code (untested):
```lua
local function set_dns_record(name, ip, sectionname)
    if not ip then return end
    uci:section('dhcp', 'domain', sectionname, {
        name = name,
        ip = ip,
    })
end

for i, name in ipairs(next_node.name or {}) do
    set_dns_record(name, next_node.ip4, 'nextnode4_' .. i)
    set_dns_record(name, next_node.ip6, 'nextnode6_' .. i)
end
```